sled-agent/src/sled_agent.rs

```diff
@@ -948,9 +948,48 @@ impl SledAgent {
         Ok(disk_result)
     }
 
+    /// Ensures that the specific sets of disks, datasets, and zones specified
+    /// by `config` are running.
+    ///
+    /// This method currently blocks while each of disks, datasets, and zones
+    /// are ensured in that order; a failure on one prevents any attempt to
+    /// ensure the subsequent step(s).
+    pub async fn set_omicron_config(
```

Comment:
Saying this explicitly, to triple-check: Because this API combines three previously `pub` methods here that each have an on-disk schema, but it calls those **same** methods exactly, we don't expect this PR to introduce any schema change, correct?

Reply:
Correct.